superset/db_engine_specs/base.py

```diff
@@ -1411,6 +1412,13 @@ def build_sqlalchemy_uri(
         encryted_extra: Optional[Dict[str, str]] = None,
     ) -> str:
         query = parameters.get("query", {})
+        if isinstance(query, str):
+            # Make sure query isn't an empty string
+            if not query:
+                query = {}
+            else:
+                query = literal_eval(query)
```

Comment:
The proper way to handle this would be fixing the frontend to always send an object, like I commented here: https://github.com/apache/superset/pull/16038/files#r681331894

We need to either fix `DatabaseObject.query` to have only type `object` [here](https://github.com/apache/superset/pull/16038/files#diff-83388bd5479bd703033c862032dd55880f32ec676024d39fd46c4a953606e517R48), or at least fix the FE so that it converts `query` correctly to an object before doing a request to the backend.